lib/adapters/MochaAdapter.js

```diff
@@ -53,60 +52,69 @@ export default class MochaAdapter extends EventEmitter {
       parent = parent.parent
     }
 
-    var test = new Test(
-      testName,
-      suiteName
-    )
-    this.idDataMap[mochaTest.title] = test
+    var test
```

Comment:
Actually a return from the `if`, then the else won't be needed anymore and be just `return new Test...`